examples/example2.cpp

```diff
@@ -3,25 +3,35 @@
 // Optionally use the namespace to do `describe` instead of `Macchiato::describe`, etc
 using namespace Macchiato;
 
-int main() {
-	describe("Foo", [&]() {
-		describe("with bar", [&]() {
-			it("should baz", [&]() {
-				return expect(true).to->equal(true)->getResult();
-			});
-
-			// This test will fail because it is false != true
-			it("should qux", [&]() {
-				return expect(false).to->equal(true)->getResult();
-			});
+class Program
+{
+	public:
+		Program()
+		{
+			foo = 1;
+		}
+		int foo;
+		int bar () { return foo - 1; }
+};
 
-			// An `it` call without a callback is considered "pending"
-			// In other words, the test still needs to written/implemented.
-			it("should norf");
+int main() {
+	describe("Program", [&]() {
+		it("should have field 'foo' with initial value 1", [&]() {
+			Program ProgramPrototype = Program ();
+			return expect(ProgramPrototype.foo).to->equal(1)->getResult();
+		});
+		it("should have function 'bar' which returns 0 in initial state", [&]() {
+			Program ProgramPrototype = Program ();
+			return expect(ProgramPrototype.bar()).to->equal(0)->getResult();
+		});
+		it("should have function 'bar' which returns 9999 when 'foo' is 1000", [&]() {
+			Program ProgramPrototype = Program ();
+			ProgramPrototype.foo = 1000;
+			return expect(ProgramPrototype.bar()).to->equal(999)->getResult();
 		});
 	});
 
 	std::cout << Macchiato::GetResultantTestOutput() << std::endl;
 
 	return 0;
-}
\ No newline at end of file
+}

```

Comment:
I don't think this adds more value over the simple version that shows off a pass, fail, pending as well.